File: LastSpool/Models/Job.cs
using System;

using System.ComponentModel.DataAnnotations;


namespace LastSpool.Models
{
    public class Job
    {
        public int Id { get; set; }
        public int PrinterId { get; set; }
        [MaxLength(255)]
        public string Image { get; set; }
        [MaxLength(255)]
        public string FileName { get; set; }
        [MaxLength(255)]
        public string StatusMessage { get; set; }
        public int PrintLength { get; set; }
        public int FilamentLength { get; set; }
        public DateTime StatusDateTime { get; set; }
        public string DeviceIdentifier { get; set; }
        
        public Printer Printer { get; set; }
        public UserProfile UserProfile { get; set; }
#nullable enable
        public int? PercentDone { get; set; }
        public int? TimeLeft { get; set; }
        public DateTime? CompleteDateTime { get; set; }
    }
}

File: LastSpool/client/src/components/Job/Job.js
import "./Job.css"
import React from "react";
import { Link, useParams } from "react-router-dom"
import { Button, Card} from "reactstrap";

const Job = ({ job }) => {

    const currentUserId = JSON.parse(sessionStorage.getItem('userProfile')).id
    const { printerId, jobId } = useParams();
    let minutesLeft = Math.floor(job.timeLeft / 60)
    let statusMessage = `Printing. The job is ${job.percentDone}% complete with an estimated ${minutesLeft} minutes remaining`

    return (
        <Card className="card mx-3 jobBox d-flex flex-column align-items-center ">

            <strong className="my-3">{job.fileName}</strong>
            <div>
                <img src={job.image} alt="job" className="jobImage" />
            </div>
            <div>{job.printer.name}</div>
            {(job.timeLeft > 0) ? statusMessage : null}

            <div className="mt-auto p-2 d-flex justify-content-center">
    


                <Link to={`/printers/${printerId}/jobs/${job.id}/details`}>
                    <Button>Details</Button>
                </Link>
                {/* {(currentUserId === job.printer.userProfileId) && (jobId) ?
                    <Link to={`/printers/${printerId}/jobs/${job.id}/delete`}>
                        <Button color="danger" className="jobButton">Delete</Button>
                    </Link>
                    : null}

                {(currentUserId === job.printer.userProfileId && (jobId)) ?
                    <Link to={`/printers/${printerId}/jobs/${job.id}/edit`}>
                        <Button className="jobButton">Edit</Button>
                    </Link>
                    : null} */}

            </div>

        </Card>
    )




}

export default Job;